web_src/js/components/DiffCommitSelector.vue

```diff
@@ -0,0 +1,191 @@
+        <div class="vertical item gt-df gt-fc gt-gap-2" @click="showAllChanges()">
+          <div class="gt-ellipsis">
+            {{ locale.show_all_commits }}
+          </div>
+          <div class="gt-ellipsis text light-2 gt-mb-0">
+            {{ locale.stats_num_commits }}
+          </div>
+        </div>
+        <!-- only show the show changes since last review if there is a review AND we are commits ahead of the last review -->
+        <div v-if="lastReviewCommitSha != null && commitsSinceLastReview > 0" class="vertical item gt-df gt-fc gt-gap-2 gt-border-secondary-top" @click="changesSinceLastReviewClick()">
+          <div class="gt-ellipsis">
+            {{ locale.show_changes_since_your_last_review }}
+          </div>
+          <div class="gt-ellipsis text light-2">
+            {{ commitsSinceLastReview }} commits
+          </div>
+        </div>
+        <span class="info gt-border-secondary-top">{{ locale.select_commit_hold_shift_for_range }}</span>
+        <template v-for="commit in commits" :key="commit.id">
+          <div class="vertical item gt-df gt-fr gt-gap-2 gt-border-secondary-top" :class="{selected: commit.selected}" @click.exact="commitClicked(commit.id)" @click.shift.exact.stop.prevent="commitClickedShift(commit)">
+            <div class="gt-f1 gt-df gt-fc gt-gap-2">
+              <div class="gt-ellipsis commit-list-summary">
+                {{ commit.summary }}
+              </div>
+              <div class="gt-ellipsis text light-2">
+                {{ commit.committer_or_author_name }}
+                <span class="text right">
+                  <relative-time class="time-since" prefix="" :datetime="commit.time" data-tooltip-content data-tooltip-interactive="true">{{ commit.time }}</relative-time>
+                </span>
+              </div>
+            </div>
+            <div class="gt-mono">
+              {{ commit.short_sha }}
+            </div>
+          </div>
+        </template>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import {SvgIcon} from '../svg.js';
+
+export default {
+  components: {SvgIcon},
+  data: () => {
+    return {
+      menuVisible: false,
+      isLoading: false,
+      locale: {},
+      commits: [],
+      hoverActivated: false,
+      lastReviewCommitSha: null
+    };
+  },
+  computed: {
+    commitsSinceLastReview() {
+      if (this.lastReviewCommitSha) {
+        return this.commits.length - this.commits.findIndex((x) => x.id === this.lastReviewCommitSha) - 1;
+      }
+      return 0;
+    },
+    queryParams() {
+      return this.$el.parentNode.getAttribute('data-queryparams');
+    },
+    issueLink() {
+      return this.$el.parentNode.getAttribute('data-issuelink');
+    }
+  },
+  mounted() {
+    // Add a click listener to close this menu on click outside of this element
+    //  when the dropdown is currently visible opened
+    document.body.addEventListener('click', (event) => {
+      if (this.$el.contains(event.target)) return;
```

Comment:
Isn't that
```suggestion
    this.$el.addEventListener('click', (event) => {
```
?

Reply:
No - this code checks for any click outside of the commitselector. Your suggestion would listen to all clicks on the element.
Please see the code comment above this line

(same logic is used in RepoBranchTagSelector.vue)

Reply:
Ah, I read that statement the wrong way around.
I read it as `only do something when the element was clicked` instead of `only do something when the element was`**not**` clicked`.